src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,166 @@
-                .withLayout(layout)
-                .withFileName(logLocation)
-                .withFilePattern(logLocation + ".%d")
-                .withPolicy(SizeBasedTriggeringPolicy.createPolicy("5MB"))
-                .withStrategy(DefaultRolloverStrategy.newBuilder().withMax("1").build())
-                .build();
-
-            fa.start();
-            config.addAppender(fa);
-            ctx.getRootLogger().addAppender(config.getAppender(fa.getName()));
-
-            log.info("File Handler set");
-        } else {
-
-            if (logLocation != null
-                    && (ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
-                        || SYSTEM_ERR_ALT.equals(logLocation))) {
-
-                ConsoleAppender console = (ConsoleAppender)config.getAppender("CONSOLE");
-                console.stop();
-                config.getRootLogger().removeAppender("CONSOLE");
-                ctx.updateLoggers();
-
-                ConsoleAppender ca = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(logLocation)
-                    .setTarget(ConsoleAppender.Target.SYSTEM_ERR)
-                    .withLayout(layout)
-                    .build();
-
-                ca.start();
-                config.addAppender(ca);
-                ctx.getRootLogger().addAppender(config.getAppender(ca.getName()));
+            @Override
+            public synchronized String format(LogRecord lr) {
+                SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat,
+                                                        Locale.getDefault());
+                return String.format(format,
+                    simpleDateFormat.format(new Date()).toString(),
+                    // NOTE(remy): these conversions may generate a lot of garbage
+                    //      (jaime): if performance impact over a sustained period of time
+                    //               isn't too bad, we can probably live with this.
+                    LogLevel.fromJulLevel(lr.getLevel()).toString(),
+                    simpleClassName(lr.getSourceClassName()),
+                    lr.getMessage()
+                );
+            }
+        };
+
+        // prepare the different handlers
+        // --
+
+        ConsoleHandler stdoutHandler = null;
+        ConsoleHandler stderrHandler = null;
+        FileHandler fileHandler = null;
+
+        // the logLocation isn't always containing a file, it is sometimes
+        // referring to a standard output. We want to create a FileHandler only
+        // if the logLocation is a file on the FS.
+        if (logLocation != null && logLocation.length() > 0) {
+            if (!isStdOut(logLocation) && !isStdErr(logLocation)) {
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, 5 * 1024 * 1024, 1);
```

Comment:
The 5mb value should be pulled up to consts

Reply:
you're right